widget/table.go

```diff
@@ -90,41 +79,51 @@ func (t *Table) CreateRenderer() fyne.WidgetRenderer {
 	return r
 }
 
-// Selection will mark the specified cell (at row, col) to be marked as selected.
-func (t *Table) Selection() (row, col int) {
-	return t.selectedRow, t.selectedColumn
+// Select will mark the specified cell as selected.
+func (t *Table) Select(id TableCellID) {
+	if f := t.OnUnselected; f != nil && t.selectedCell != nil {
+		f(*t.selectedCell)
+	}
+	t.selectedCell = &id
+
+	t.scrollTo(t.selectedCell)
+	if t.moveCallback != nil {
+		t.moveCallback()
+	}
+
+	if f := t.OnSelected; f != nil {
+		f(id)
+	}
 }
 
-// SetSelection will mark the specified cell (at row, col) to be marked as selected.
-func (t *Table) SetSelection(row, col int) {
-	t.selectedRow = row
-	t.selectedColumn = col
+// Unselect will mark the cell provided by id as unselected.
+func (t *Table) Unselect(id TableCellID) {
```

Comment:
Similar optimization here, if t.selectedCell is already nil return early

Reply:
Resolved